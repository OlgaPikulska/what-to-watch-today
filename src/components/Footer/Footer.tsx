import { Typography } from "@mui/material";
import { FC } from "react";

export const Footer: FC = () => {
	const currentYear: number = new Date().getFullYear();
	return (
		<footer className="bg-footer p-4 text-center">
			<Typography variant="subtitle1" color="secondary.dark">
				{`Â© ${currentYear} | All Rights Reserved | Created by`}{" "}
				<span className="font-bold">Olga Pikulska</span>
			</Typography>
		</footer>
	);
};
