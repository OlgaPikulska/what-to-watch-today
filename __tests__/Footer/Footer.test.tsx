import { Footer } from "@/components/Footer/Footer";
import { render } from "@testing-library/react";

describe("Footer Component", () => {
	it("should match the snapshot", () => {
		const { container } = render(<Footer />);
		expect(container).toMatchSnapshot();
	});

	it("should render current year", () => {
		const currentYear = new Date().getFullYear().toString();
		const { getByText } = render(<Footer />);

		expect(getByText((content) => content.includes(`Â© ${currentYear}`))).toBeInTheDocument();
	});
});
