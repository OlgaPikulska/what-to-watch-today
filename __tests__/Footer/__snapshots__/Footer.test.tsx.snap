// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match the snapshot 1`] = `
<div>
  <footer
    class="bg-footer p-4 text-center"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-1i9ek8c-MuiTypography-root"
    >
      Â© 2025 | All Rights Reserved | Created by
       
      <span
        class="font-bold"
      >
        Olga Pikulska
      </span>
    </h6>
  </footer>
</div>
`;
